services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImpl.java

```diff
@@ -179,8 +190,29 @@ public OIDCAuthorizationRequestDescription getAuthenticationRequestDescription(
 		}
 		return scopeDescriptions;
 	}
+	
+	static String getScopeHash(OIDCAuthorizationRequest authorizationRequest) {
+		String text = authorizationRequest.getScope()+authorizationRequest.getClaims();
+		MessageDigest digest;
+		try {
+			digest = MessageDigest.getInstance("SHA-256");
+		} catch (NoSuchAlgorithmException e) {
+			throw new RuntimeException(e);
+		}
+		byte[] hash = digest.digest(text.getBytes(StandardCharsets.UTF_8));
+		return new String(hash, StandardCharsets.UTF_8);
+	}
+	
+	@Override
+	public boolean hasUserGrantedConsent(UserInfo userInfo, OIDCAuthorizationRequest authorizationRequest) {
+		Date grantedOn = oauthDao.lookupAuthorizationConsent(userInfo.getId(), 
```

Comment:
This whole logic can go in the dao and the dao method can take in input the threshold